Add a method to temporarily disable easyescape

fwSmit opened the pull request:
I've added a method to disable easyescape. This is useful when you want to have easyescape on for certain filetypes and off for others. Or you can disable it when having spell language set to Dutch.
That last one is useful to me, because I like having easyescape bound to 'jk', but that doesn't work well when typing in Dutch. Changing the mapping doesn't work, because mappings have to be regenerated, so I found this to be an easy option.

zhou13 commented:
Thank you for adding this feature. I think it is better to implement it with a local buffer variable like `b:easyescape_disabled`. If we need to disable easyescape on a buffer, we just need to set it without worry about resetting it when we enter another buffer.

BTW, `jk` is conflicting with words in Dutch, it might be better to use other bindings?

fwsmit commented:
Right that makes it a whole lot easier. I've implemented that and corrected the docs accordingly. I've also added a modeline at the end, to make sure the formatting isn't different when you have expandtab disabled in your vimrc.

As for using another shortcut, that is certainly an option, but I really like the jk shortcut and I don't really type a whole lot of dutch in vim.

zhou13 commented:
This looks good to me. Thank you for the pull request!
